sanitychecks.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+function verifyWorkingDirIsWorldReadable {
+  if [ ! -d $WORKING_DIR ]; then
+    echo "WORKING_DIR is not a directory"
+    exit 1
+  fi
+
+  if [ ! -r $WORKING_DIR ]; then
+    echo "Unable to access WORKING_DIR"
+    exit 1
+  fi
+
+  ATTRIBS=$(stat -L -c "0%a" "$WORKING_DIR")
+
+  if (( ($ATTRIBS & 05) != 0 )); then
+
+   WD_PATH=$(dirname $WORKING_DIR)
+   while [ "$WD_PATH" != "/" ] ; do
```

Comment:
I think this function can be simplified, how about something like:

```
if find "${WORKING_DIR}" -maxdepth 1 ! -perm -o=r | grep -q . ; then
    echo "The WORKING_DIR ${WORKING_DIR}" is not world-readable!"
    exit 1
fi
```

Reply:
Seems a good idea, I tested it but I had some troubles with the `maxdepth` option, since it fails whenever the target dir already contains (for any reason) a sub directory not readable by others. Instead with `-maxdepth 0` works fine also in such cases.

Reply:
Hi @hardys, I've retested the approach and found a case where it seems to not work properly when WORKING_DIR is placed within the _current user_ home directory (ie, user `andrea` and `WORKING_DIR=/home/andrea/wd") and `wd` is flagged as world-readable - while `/home/andrea' is not. Not sure how to cover it case with `find`, previous approach was fine also in such case.